### DISCUSSION TITLE
How to custom a component in v9

### DISCUSSION THREAD

**Discussion by tienlx97** [microsoft/fluentui#28312] [Q&A]
Example I want custom Checkbox style. Here is example:

```tsx

import { CheckboxProps, Checkbox, makeStyles } from '@fluentui/react-components'
import React from 'react'

export default function SLCheckbox({ label, ...props }: CheckboxProps) {
  // eslint-disable-next-line @typescript-eslint/no-use-before-define
  const classes = useStyles()
  return (
    <Checkbox
      {...props}
      indicator={{
        className: classes.large,
      }}
      label={{
        className: classes.label,
        ...label, // Error here
      }}
      size="large"
    />
  )
}

const useStyles = makeStyles({
  large: {
    width: '24px',
    height: '24px',
  },

  label: {
    paddingLeft: '0px',
  },
})
```

Why the label error, I don't know why ?

**Comment by spmonahan** [comment]
I believe the error you are seeing is because spread for generic types (`label` is a generic) is not yet supported in Typescript ([1](https://github.com/Microsoft/TypeScript/issues/10727), [2](https://stackoverflow.com/questions/51189388/typescript-spread-types-may-only-be-created-from-object-types)).

Even if you tried a cast or switched to `Object.assign` rather than spread you'll still run into issues since `label` is a slot it may also be a string or `null` or `undefined` so you'll have to handle all those possibilities as well.

One way to handle this is to recompose the checkbox: https://codesandbox.io/s/label-customization-j3dqpg
